src/Planner/PlannerExpressionAnalysis.cpp

```diff
@@ -109,7 +109,7 @@ std::optional<AggregationAnalysisResult> analyzeAggregation(const QueryTreeNodeP
                             continue;
 
                         auto expression_type_after_aggregation = group_by_use_nulls ? makeNullableSafe(expression_dag_node->result_type) : expression_dag_node->result_type;
-                        available_columns_after_aggregation.emplace_back(nullptr, expression_type_after_aggregation, expression_dag_node->result_name);
```

Comment:
I'm not sure that sharing const column for aggregator is a good idea, since there are some manipulations to it, so I think it is better to avoid sharing this column at all, like here https://github.com/ClickHouse/ClickHouse/pull/59986/files#diff-06a4eb7b985c6d6bcb43e37d94ada33f85a846a26674641e6648395f704bcc23R96

Reply:
I compared to current version's DAG for the case in this PR and seems it is also sharing this column...